services/net-service.ts

```diff
@@ -71,6 +77,53 @@ export class Net implements INet {
 
 		return startPort;
 	}
+
+	public async waitForPortToListen(port: number, timeout: number, interval?: number): Promise<boolean> {
+		interval = interval || Net.DEFAULT_INTERVAL;
+		const endTime = new Date().getTime() + timeout;
+		const platformData: IDictionary<any> = {
+			"darwin": {
+				command: "netstat -f inet -p tcp -anL",
+				regex: new RegExp(`\\.${port}\\b`, "g")
+			},
+			"linux": {
+				command: "netstat -tnl",
+				regex: new RegExp(`:${port}\\s`, "g")
+			},
+			"win32": {
+				command: "netstat -ant -p tcp",
+				regex: new RegExp(`TCP\\s+(\\d+\\.){3}\\d+:${port}.*?LISTEN`, "g")
+			}
+		};
+
+		const platform = this.$osInfo.platform();
+		const currentPlatformData = platformData[platform];
+		if (!currentPlatformData) {
+			this.$errors.failWithoutHelp(`Unable to check for free ports on ${platform}. Supported platforms are: ${_.keys(platformData).join(", ")}`);
+		}
+
+		while (true) {
```

Comment:
Instead of `while (true)` you can set the condition `while (new Date().getTime() < endTime)` and delete the `break;` code below.

IMO it'd be a bit clearer than looking at what seems to be an endless loop at a first glance

Reply:
The current flow executes at least one check, no matter of the passed timeout. Changing the `while` statement will change this behavior.
Also the current workflow checks if a timeout is passed and performs at least one check after that. Changing the `while` statement will change this behavior as well, as we have `sleep` at the end of the `while`. The `sleep` cannot be moved at the first line of the `while` as we may not have to wait for it.
So the desired workflow is:
1. Check if the port is in LISTEN state.
2. If yes - break. If not, check if timeout is passed and in case not - sleep for specified interval. 
3. Check if port is in LISTEN state.
4. If yes - break. If not, check if timeout is passed and in case not - sleep for specified interval.
... Repeat until timeout is reached.